src/RemoteAPI/class-remote-api-base.php

```diff
@@ -151,8 +153,13 @@ public function get_api_url( array $query ): string {
 	 */
 	public function get_items( $query, $associative = false ) {
 		$full_api_url = $this->get_api_url( $query );
-		$options      = $this->get_request_options();
-		$response     = wp_safe_remote_get( $full_api_url, $options );
+		/**
+		 * GET request options.
+		 *
+		 * @var WP_HTTP_Request_Args $options
+		 */
```

Comment:
Instead of adding this as comment I think we can improve type of `get_request_options` function.

Reply:
Unless I'm doing something wrong, adding PHPStan types as `@return` comments have significant negative side-effects in my experience. Let's take the following function for example:

```php
	/**
	 * Returns the request's options for the remote API call.
	 *
	 * @since 3.9.0
	 *
	 * @return array<string, mixed> The array of options.
	 */
	protected function get_request_options(): array {
		return array();
	}
```

Now let's add the PHPStan type to the `@return` comment:

```php
	/**
	 * Returns the request's options for the remote API call.
	 *
	 * @since 3.9.0
	 *
	 * @return WP_HTTP_Request_Args The array of options.
	 */
	protected function get_request_options(): array {
		return array();
	}
```

We now got a PHPStan error:
```
Method Parsely\RemoteAPI\Remote_API_Base::get_request_options() should return array{method: string, timeout: float, redirection: int, httpversion: string, user-agent: string, reject_unsafe_urls: bool, blocking: bool, headers: array<string, string>, ...} but returns array{}.
```

OK, let's remove the function's return type then:
```php
	/**
	 * Returns the request's options for the remote API call.
	 *
	 * @since 3.9.0
	 *
	 * @return WP_HTTP_Request_Args The array of options.
	 */
	protected function get_request_options() {
		return array();
	}
```

We now get 2 PHPStan errors:
```
Out of 5 possible return types, only 60 % actually have it. Add more return types to get over 94 %.
Method Parsely\RemoteAPI\Remote_API_Base::get_request_options() should return array{method: string, timeout: float, redirection: int, httpversion: string, user-agent: string, reject_unsafe_urls: bool, blocking: bool, headers: array<string, string>, ...} but returns array{}.
```

OK, let's add a `@var` comment then:
```php
	/**
	 * Returns the request's options for the remote API call.
	 *
	 * @since 3.9.0
	 *
	 * @return WP_HTTP_Request_Args The array of options.
	 */
	protected function get_request_options() {
		/**
		 * GET request options.
		 *
		 * @var WP_HTTP_Request_Args
		 */
		return array();
	}
```

We're still getting the `Out of 5 possible return types, only 60 % actually have it. Add more return types to get over 94 %.` error though. And we need to perform this manipulation to other functions that are now incompatible with this one. We can try adding `: array` back to the function, but this only leads to other errors.

The conclusion is that not only our code has become bulkier, but also our global return type coverage has lowered, and we would need to lower our global return type coverage in `phpstan.neon`. This is why:
- I'm totally against using PHPStan types as parameter or return types.
- I'm totally in favor of using normal PHP types because of the many benefits they bring, including type recognition and parameter/return type coverage increase.

Please let me know if you think I'm missing something.

Reply:
> Method Parsely\RemoteAPI\Remote_API_Base::get_request_options() should return array{method: string, timeout: float, redirection: int, httpversion: string, user-agent: string, reject_unsafe_urls: bool, blocking: bool, headers: array<string, string>, ...} but returns array{}.

It seems like we can marked the fields as optional in PHPStan type to avoid this issue.

Reply:
Experimented with this a bit more, but couldn't get to a state without errors. So keeping as is, but happy to re-discuss about it in the future if we think it's important.